api/tests/python/services/name_request/test_abstract_name_request.py

```diff
@@ -0,0 +1,44 @@
+"""Tests for AbstractNameRequestMixin."""
+import datetime
+
+import pytest
+from dateutil.tz import gettz
+
+from namex.services.name_request import NameRequestService
+from namex.models import Request as RequestDAO
+
+nr_svc = NameRequestService()
+pacific_tz = gettz("UTC")
+
+
+@pytest.mark.parametrize("input_datetime,expected_date", [
+    (datetime.datetime(2021, 1, 28, 14, 11, 00, tzinfo=pacific_tz), datetime.date(2021, 3, 25)),
+    (datetime.datetime(2021, 2, 28, 14, 11, 00, tzinfo=pacific_tz), datetime.date(2021, 4, 25)),
+    (datetime.datetime(2021, 3, 28, 14, 11, 00, tzinfo=pacific_tz), datetime.date(2021, 5, 23)),
+    (datetime.datetime(2021, 4, 28, 14, 11, 00, tzinfo=pacific_tz), datetime.date(2021, 6, 23)),
+    (datetime.datetime(2021, 5, 28, 14, 11, 00, tzinfo=pacific_tz), datetime.date(2021, 7, 23)),
+    (datetime.datetime(2021, 6, 28, 14, 11, 00, tzinfo=pacific_tz), datetime.date(2021, 8, 23)),
+    (datetime.datetime(2021, 7, 28, 14, 11, 00, tzinfo=pacific_tz), datetime.date(2021, 9, 22)),
+    (datetime.datetime(2021, 8, 28, 14, 11, 00, tzinfo=pacific_tz), datetime.date(2021, 10, 23)),
+    (datetime.datetime(2021, 9, 28, 14, 11, 00, tzinfo=pacific_tz), datetime.date(2021, 11, 23)),
+    (datetime.datetime(2021, 10, 28, 14, 11, 00, tzinfo=pacific_tz), datetime.date(2021, 12, 23)),
+    (datetime.datetime(2021, 11, 28, 14, 11, 00, tzinfo=pacific_tz), datetime.date(2022, 1, 23)),
+    (datetime.datetime(2021, 12, 28, 14, 11, 00, tzinfo=pacific_tz), datetime.date(2022, 2, 22))
+])
+def test_create_expiry_date(input_datetime, expected_date):
+    """Test that create_expiry_date method returns a datetime at added X days and at 11:59pm.
+
+    It should not change the time (hour and minute) because of TimeZone offset changes through the year.
+    """
+    expiry_date = nr_svc.create_expiry_date(input_datetime, 56)
+    assert expiry_date
+    assert expiry_date.time().hour == 23
+    assert expiry_date.time().minute == 59
+    assert expiry_date.strftime('%z') in ['-0800', '-0700'] # should not be +0000 or any other
```

Comment:
This should specifically look for a 8h offset during Standard time or a 7h offset during Daylight time. This can be a test parameter using known clock change values.

Ref: https://in-the-sky.org/timezone.php

Reply:
yes, I know but we should keep track of the dates that it changes. The python libraries do their jobs and the date looks to be calculated correctly at this point.

Reply:
outdated comment - closing